EEJT-000A/main/java/task2/SearchByDateChange.java

```diff
@@ -0,0 +1,65 @@
+package task2;
+
+import java.io.File;
+import java.sql.Timestamp;
+import java.text.SimpleDateFormat;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class SearchByDateChange implements SearchByParam {
+
+    private SearchByParam searchByParam;
+    private final String Patt = "yyyy-MM-dd HH:mm:ss";
+
+    public SearchByDateChange(SearchByParam searchByParam) {
+        this.searchByParam = searchByParam;
+    }
+    private boolean isNextChain (SearchByParam searchByParam){
+        return searchByParam != null;
+    }
+    private boolean isFileMatches(File file , Parameters parameters){
+        return file.lastModified() <= Timestamp.valueOf(parameters.getDateMore()).getTime()
+                && file.lastModified() >= Timestamp.valueOf(parameters.getDateLess()).getTime() ;
+    }
+
+    @Override
+    public List<String> search(Parameters parameters, List<String> list) {
```

Comment:
Some of this method logic is repeated in other `SearchByParam`  classes. Please think how to remove this duplication.
And also this method looks pretty big to me. Try to follow the rule ">20 lines in method -> split it using private method or any other means".